### ISSUE TITLE
Linear interpolation resampling causes severe audio quality degradation

### ISSUE THREAD

**Issue by tassa-yoniso-manasi-karoto** [Ryan5453/demucs-next#4]
## Summary

When input audio sample rate differs from the model's expected 44.1kHz, `convert_audio()` uses `torch.nn.functional.interpolate` with `mode="linear"`, which produces extremely poor quality output.

## Reproduction

1. Use any audio file at 48kHz
2. Run separation with htdemucs
3. Compare output quality vs same file pre-resampled to 44.1kHz using a proper resampler (ffmpeg soxr)

The 48kHz input sounds heavily degraded, like extreme lossy compression.

## Root Cause

`demucs/audio.py` lines 58-61:

```python
# Use linear interpolation for audio resampling
resampled = F.interpolate(
    wav_reshaped, scale_factor=ratio, mode="linear", align_corners=False
)
```

Linear interpolation simply draws straight lines between samples. For audio resampling, this:
- Introduces aliasing artifacts
- Causes high-frequency loss
- Produces "muddy" or "compressed" sounding output

Proper audio resampling requires sinc interpolation with anti-aliasing filters.

## Claude Opus Suggested Fix

Replace with `torchaudio.functional.resample` or `torchaudio.transforms.Resample`:

```python
import torchaudio.functional as F_audio

def convert_audio(wav, from_samplerate, to_samplerate, channels) -> Tensor:
    wav = convert_audio_channels(wav, channels)

    if from_samplerate == to_samplerate:
        return wav

    return F_audio.resample(wav, from_samplerate, to_samplerate)
```

This uses a polyphase Kaiser window resampler which is standard for audio.

## Note

The old `predict.py` (deleted in commit 497f3af "Demucs Beta") used Meta's official demucs which handled resampling correctly. The Replicate deployment still uses that older version, which is why Replicate produces (or at least produced AFAIK) correct quality output while local demucs-next installations using the new code do not.

<hr>

Thanks for making this repo!!
